### PR TITLE
[WFCORE-7108] Add a ModuleIdentifierUtil method to parse a module ide…

### PR DESCRIPTION
**Pull request by bstansberry**
…ntifier into name and slot and process the tuple

https://issues.redhat.com/browse/WFCORE-7108

Needed for https://github.com/wildfly/wildfly/pull/18561

### CODE REVIEW

File: controller/src/main/java/org/jboss/as/controller/ModuleIdentifierUtil.java [hunk 1]
```diff
@@ -21,32 +23,88 @@ public final class ModuleIdentifierUtil {
...
+
+
+    /**
+     * Parses the given module identifier into name and optional slot elements, passing those to the given
+     * function and returning the result of that function.
+     * <p/>
+     *
+     * @param moduleIdentifier an identifier for a module. Cannot be {@code null}
+     * @param function a function to apply to the module's name and optional slot. Cannot be {@code null}.
+     *                 The slot value passed the function may be null if the identifier does not contain one.
```
Comment (ropalka) [inline controller/src/main/java/org/jboss/as/controller/ModuleIdentifierUtil.java:53]: "passed the function"  should be "passed to the function"

File: controller/src/main/java/org/jboss/as/controller/ModuleIdentifierUtil.java [hunk 2]
```diff
@@ -21,32 +23,88 @@ public final class ModuleIdentifierUtil {
...
```
Comment (ropalka) [inline controller/src/main/java/org/jboss/as/controller/ModuleIdentifierUtil.java:70]: same as above

**Review by ropalka — APPROVED** [review]
Except minor wording issue in javadoc it looks good to me.

File: controller/src/main/java/org/jboss/as/controller/ModuleIdentifierUtil.java (hunk 2 above)
Comment (bstansberry) [reply controller/src/main/java/org/jboss/as/controller/ModuleIdentifierUtil.java]: Thanks; fixed.

File: controller/src/main/java/org/jboss/as/controller/ModuleIdentifierUtil.java (hunk 1 above)
Comment (bstansberry) [reply controller/src/main/java/org/jboss/as/controller/ModuleIdentifierUtil.java]: Thanks; fixed.

**Pull request merged**